### PR TITLE
Refactoring City, changes to Tile  inheritance hierarchy, and GUI fix.

### PR DESCRIPTION
**Pull request by jacquelinedseal**
Refactored City to swap color attribute out for 'block' attribute (an int indicating a particular colour group), since the application logic doesn't rely on colour directly and specific formatting decisions should be made in the interface adapters layer. Adjusted Director accordingly. Added method in Board to package entity data into a DTO. Refactored Tile inheritance hierarchy to accommodate upcoming PR for GUI fix: removed redundant SpecialTile (JailTile, AuctionTile etc now inherit directly from Tile) and added isOwned() and getName() methods to Tile.

### CODE REVIEW

**Review by samrooj — APPROVED** [review]
looks good!

File: src/main/java/entities/Board.java [hunk 1]
```diff
@@ -26,4 +31,40 @@ public Tile getTileAt(int index){
...
+                dto = new TileDTO(name, i, false, owner, 0, rent, 8);
+            } else {
+                dto = new TileDTO(name, i, false, owner, 0, 0, 9);
+            }
+            boardData.put(name, dto);
+        }
+
+        return boardData;
+    }
 }
```
Comment (stevechoi0222) [inline src/main/java/entities/Board.java:70]: Looks good!

File: src/main/java/interfaceadapters/BoardPanel.java [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+     */
+    private void drawTokens(int numOfPlayers, Map<Integer, Point2D> tokenPositions,
+                            Map<Integer, Color> playerToColour){
+        for(int i = 1; i <= numOfPlayers; i++){
+            Point2D position = tokenPositions.get(i);
+            List<JComponent> tileComponent = this.positionToTile.get(position);
+            tileComponent.get(i - 1).setBackground(playerToColour.get(i));
+        }
+    }
+}
```
Comment (stevechoi0222) [inline src/main/java/interfaceadapters/BoardPanel.java:217]: This is a killer

**Review by stevechoi0222 — APPROVED** [review]
Crazy work. Awesome job!

**Review by parthvats02 — APPROVED** [review]
The changes look good to me!

**Pull request merged**